pkg/controller/infra/infra_controller.go

```diff
@@ -201,18 +366,145 @@ func (c *InfraController) handleErr(err error, key interface{}) {
+		}
+		if clusterProvisioned == nil {
+			cluster.Status.Conditions = append(cluster.Status.Conditions, clusteroperator.ClusterCondition{
+				Type:               clusteroperator.ClusterHardwareProvisioned,
+				Status:             kapi.ConditionTrue,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             "JobCompleted",
+				Message:            fmt.Sprintf("Job %s/%s completed at %v", job.Namespace, job.Name, jobCompleted.LastTransitionTime),
+			})
+		}
+		provisioningFailed := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioningFailed)
+		if provisioningFailed != nil &&
+			provisioningFailed.Status == kapi.ConditionTrue {
+			provisioningFailed.Status = kapi.ConditionFalse
+			provisioningFailed.LastTransitionTime = now
+			provisioningFailed.LastProbeTime = now
+			provisioningFailed.Reason = ""
+			provisioningFailed.Message = ""
+		}
+		cluster.Status.Provisioned = true
+		cluster.Status.ProvisioningJob = nil
+	case jobFailed != nil && jobFailed.Status == kapi.ConditionTrue:
+		clusterProvisioning := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioning)
+		if clusterProvisioning != nil &&
+			clusterProvisioning.Status == kapi.ConditionTrue {
+			clusterProvisioning.Status = kapi.ConditionFalse
+			clusterProvisioning.LastTransitionTime = now
+			clusterProvisioning.LastProbeTime = now
+			clusterProvisioning.Reason = "JobFailed"
+			clusterProvisioning.Message = fmt.Sprintf("Job %s/%s failed at %v, reason: %s", job.Namespace, job.Name, jobFailed.LastTransitionTime, jobFailed.Reason)
+		}
+		cluster.Status.ProvisioningJob = nil
+		provisioningFailed := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioningFailed)
+		if provisioningFailed != nil {
+			provisioningFailed.Status = kapi.ConditionTrue
+			provisioningFailed.LastTransitionTime = now
+			provisioningFailed.LastProbeTime = now
+			provisioningFailed.Reason = "JobFailed"
+			provisioningFailed.Message = fmt.Sprintf("Job %s/%s failed at %v, reason: %s", job.Namespace, job.Name, jobFailed.LastTransitionTime, jobFailed.Reason)
+		} else {
+			cluster.Status.Conditions = append(cluster.Status.Conditions, clusteroperator.ClusterCondition{
+				Type:               clusteroperator.ClusterHardwareProvisioningFailed,
+				Status:             kapi.ConditionTrue,
+				LastProbeTime:      now,
+				LastTransitionTime: now,
+				Reason:             "JobFailed",
+				Message:            fmt.Sprintf("Job %s/%s failed at %v, reason: %s", job.Namespace, job.Name, jobFailed.LastTransitionTime, jobFailed.Reason),
+			})
+		}
+	default:
+		clusterProvisioning := clusterCondition(cluster, clusteroperator.ClusterHardwareProvisioning)
+		reason := "JobRunning"
+		message := fmt.Sprintf("Job %s/%s is running since %v. Pod completions: %d, failures: %d", job.Namespace, job.Name, job.Status.StartTime, job.Status.Succeeded, job.Status.Failed)
```

Comment:
I will remove the pod info from the message. We'd said that we wouldn't update the running condition every time, since it would just cause a hot loop of updates.

Reply:
If we only update it when the message changes, it won't hot loop. It would be nice to see this information in the condition. I'm fine with either way--removing the info or updating the info.